micrometer-core/src/main/java/io/micrometer/core/instrument/binder/httpcomponents/MicrometerHttpRequestExecutor.java

```diff
@@ -121,25 +130,96 @@ public HttpResponse execute(HttpRequest request, HttpClientConnection conn, Http
             throw e;
         }
         finally {
-            Iterable<Tag> tags = Tags.of(extraTags).and(routeTags).and(uri, method, status);
+            Tags routeTags = exportTagsForRoute ? HttpContextUtils.generateTagsForRoute(context) : Tags.empty();
+            Tags otherTags = Tags.of("method", request.getRequestLine().getMethod(), "uri", uriMapper.apply(request),
+                    "status", status.getValue());
 
-            timerSample.stop(Timer.builder(METER_NAME).description("Duration of Apache HttpClient request execution")
-                    .tags(tags).register(registry));
+            sample.stop(otherTags.and(routeTags).and(extraTags));
```

Comment:
So if we're using the Observation API we don't want to set these tags? I'm just wondering if it should matter - shouldn't we just set all the tags upfront?

Reply:
If we're using the Observation API, those tags are set by the default convention. I don't like the way this currently works because the Tag(s) objects are made even if we aren't using them.